domain/port/state/types.go

```diff
@@ -30,6 +30,24 @@ func (pr portRange) decode() network.PortRange {
 	}
 }
 
+// unitNamePortRange represents a range of ports for a given protocol for a
+// given unit.
+type unitNamePortRange struct {
+	UnitName string `db:"name"`
+	Protocol string `db:"protocol"`
+	FromPort int    `db:"from_port"`
+	ToPort   int    `db:"to_port"`
+}
+
+// decode returns the network.PortRange representation of the unitNamePortRange.
+func (upr unitNamePortRange) decode() network.PortRange {
```

Comment:
This shouldn't be done here. I do want to fix this before the epic is done. State should 💯 not know about core.